### ISSUE TITLE
Middlewares not chained properly when using func(*Context)error

### ISSUE THREAD

**Issue by Slacki** [gernest/utron#50]
# Lets start with working example

There are three middleware functions of type `func(http.Handler)http.Handler` defined as follows:
```
func middleware_classic_1(h http.Handler) http.Handler {
	fmt.Println("md_classic_1")
	return h
}
func middleware_classic_2(h http.Handler) http.Handler {
	fmt.Println("md_classic_2")
	return h
}
func middleware_classic_3(h http.Handler) http.Handler {
	fmt.Println("md_classic_3")
	return h
}
```

The middlewares are then passed to `utron.RegisterController`
```
func init() {
	utron.RegisterController(&Index{
		Routes: []string{
			"get;/;Home",
			"post;/login;Login",
		}}, middleware_classic_1, middleware_classic_2, middleware_classic_3)
}
```

The output of this piece of code is correct, that is
![image](https://cloud.githubusercontent.com/assets/3284061/14966607/a1aa4d6a-10b3-11e6-9090-5fe971c5344d.png)

# Now exactly the same setup, but with middlewares of type `func(*Context)error`

```
func middleware1(c *utron.Context) error {
	fmt.Println("md1")
	return nil
}
func middleware2(c *utron.Context) error {
	fmt.Println("md2")
	return nil
}
func middleware3(c *utron.Context) error {
	fmt.Println("md3")
	return nil
}
```

And they are registered like this
```
func init() {
	utron.RegisterController(&Index{
		Routes: []string{
			"get;/;Home",
			"post;/login;Login",
		}}, middleware1, middleware2, middleware3)
}
```

But this time output looks like this:
![image](https://cloud.githubusercontent.com/assets/3284061/14966690/059973c8-10b4-11e6-8ad5-72d477845039.png)

## In conclusion (TL;DR)

When there are three middlewares of type `func(*Context)error` registered with `Controller`, they are not chained up correctly. Instead the last one is executed three times.

Didn't try to combine those two types of middlewares together.

```
$ go version
>>> go version go1.5.2 windows/amd64
$ ver
>>> Microsoft Windows [Version 10.0.10586]
```

**Comment by gernest** [comment]
@Slacki Thanks for reporting. I will check this later.
